src/cordova/restore-util.js

```diff
@@ -111,7 +110,7 @@ function installPlatformsFromConfigXML (platforms, opts) {
 
         // No platforms to restore from either config.xml or package.json.
         if (comboArray.length <= 0) {
-            return Q('No platforms found in config.xml or package.json. Nothing to restore');
+            return Promise.resolve('No platforms found in config.xml or package.json. Nothing to restore');
```

Comment:
This pattern seems awkward to me, since it feels more like these should be emitted as logging events.
Does this function always resolve with a log message string, or with an array of the added platforms? Currently it does both and it would be better in terms of API to be consistent with the resolved value type.